src/AggregateFunctions/AggregateFunctionQuantileGK.cpp

```diff
@@ -261,11 +260,15 @@ class ApproxSampler
 
         size_t sampled_len = 0;
         readBinaryLittleEndian(sampled_len, buf);
+        if (sampled_len > compress_threshold)
```

Comment:
For this to work we need to also check `compress_threshold` against `default_compress_threshold` (which we should probably do anyway).

Reply:
I can't push to the branch directly to fix it so here is what I was planning on adding:

```patch
diff --git a/src/AggregateFunctions/AggregateFunctionQuantileGK.cpp b/src/AggregateFunctions/AggregateFunctionQuantileGK.cpp
index de8ab65982f..f991282fbb1 100644
--- a/src/AggregateFunctions/AggregateFunctionQuantileGK.cpp
+++ b/src/AggregateFunctions/AggregateFunctionQuantileGK.cpp
@@ -17,6 +17,7 @@ namespace DB
 namespace ErrorCodes
 {
     extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    extern const int INCORRECT_DATA;
     extern const int LOGICAL_ERROR;
     extern const int NOT_IMPLEMENTED;
 }
@@ -255,6 +256,12 @@ public:
     void read(ReadBuffer & buf)
     {
         readBinaryLittleEndian(compress_threshold, buf);
+        if (compress_threshold != default_compress_threshold)
+            throw Exception(
+                ErrorCodes::INCORRECT_DATA,
+                "The compress threshold ({}) isn't the expected one ({})",
+                compress_threshold,
+                default_compress_threshold);
         readBinaryLittleEndian(relative_error, buf);
         readBinaryLittleEndian(count, buf);
 ```